client/source/js/modules/analysis/optimization.html

```diff
@@ -1,370 +1,368 @@
 <div class="page">
+  <p ng-show="needData===true" class="error-hint">Please upload Optima spreadsheet first.</p>
```

Comment:
ng-show already expects condition to be truthy, just ng-show="needData" is enough and verbose